.github/workflows/master.yml

```diff
@@ -0,0 +1,279 @@
+          cargo build --release -p witnet -p witnet_toolkit
+
+      # Confirm openssl compiled staticly
+      - name: Confirm openssl compiled staticly
+        run: LDD.exe target\release\witnet.exe | Select-String -Pattern "ssl"
+
+      # Upload Build Releases
+      - uses: actions/upload-artifact@v3
+        with:
+          name: windows-release
+          path: |
+            target\release\witnet.exe
+            target\release\witnet_toolkit.exe
+
+#?####################################################################################################?#
+#?                                                                                                    ?#
+#?                                        Sign & Publish                                              ?#
+#?                                                                                                    ?#
+#?####################################################################################################?#
+
+  Sign:
+    needs: [Build_windows, Build_macOs, Build_x86_64, Build_armv7, Build_aarch64]
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout Repo
+      - uses: actions/checkout@v3
+
+      # Download Releases
+      - uses: ./.github/actions/download-releases
+        name: Downloading Artifacts
+
+      # Import GPG Key
+      - name: Import GPG key
+        id: import_gpg
+        uses: crazy-max/ghaction-import-gpg@v5
+        with:
+          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.GPG_PASSPHRASE  }}
+      
+      # Trust GPG key
+      - name: Trust GPG key
+        run: |
+          gpg --no-tty --command-fd 0 --edit-key info@witnet.foundation << EOTRUST
+          trust
+          5
+          y
+          quit
+          EOTRUST
+      - name: Sign Hashes
+        run: |
+          cd release
+          gpg --output SHA256SUMS.asc --default-key info@witnet.foundation --detach-sig --clearsign SHA256SUMS && rm SHA256SUMS
+      
+      # Artifact Final Release Files
+      - uses: actions/upload-artifact@v3
+        with:
+          name: final-release
+          path: release/
+
+  Release:
+    needs: [Sign]
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      # Checkout Repo
+      - uses: actions/checkout@v3
+
+      # Download Release Files
+      - uses: actions/download-artifact@v3
+        with:
+          name: final-release
+          path: release/
+      
+      # Replace tag in RELEASE.md
+      - run: |
+          TAG=${{github.ref_name}}
+          sed -i "s/LATEST_VERSION/$TAG/g" RELEASE.md
+
+      # Publish new github release
+      - uses: ncipollo/release-action@v1
+        with:
+          name: "Witnet-rust ${{github.ref_name}}"
+          artifacts: "./release/*"
+          bodyFile: "RELEASE.md"
+          artifactErrorsFailBuild: true
+    
+  Publish:
```

Comment:
Is there a way we can publish it as a pre-release?

Then we only need to fill in the release notes and manually turn it into a proper release once all that info is in place.

Otherwise someone could see the notification for the new release, go to the release page and find that there's no info.

Reply:
we can, problem is, if it's a pre-release and i fetch "latest binaries" in the following stage for building the docker images will they be outdated or will it fetch from a pre-release

Reply:
They will be outdated. Ok, let's leave it as is!